libs/utils/decorate.js

```diff
@@ -1,29 +1,88 @@
 import { decorateLinkAnalytics } from '../martech/attributes.js';
+import { createTag, getConfig, } from '../utils/utils.js';
 
 export function decorateButtons(el, size) {
   const buttons = el.querySelectorAll('em a, strong a');
   if (buttons.length === 0) return;
-  buttons.forEach((button) => {
+  buttons.forEach((button, idx) => {
     const parent = button.parentElement;
-    const buttonType = parent.nodeName === 'STRONG' ? 'blue' : 'outline';
+    let buttonType = parent.nodeName === 'STRONG' ? 'blue' : 'outline';
+    if (buttonType === 'outline' && button.firstChild.nodeName === 'STRONG') buttonType = 'fill';
     button.classList.add('con-button', buttonType);
     if (size) button.classList.add(size); /* button-L, button-XL */
-    parent.insertAdjacentElement('afterend', button);
-    parent.remove();
+    const actionArea = button.closest('p');
+    actionArea?.classList.add('action-area');
+    if (idx === buttons.length - 1) {
+      button.closest('p')?.nextElementSibling?.classList.add('supplemental-text', 'body-XL');
+    }
   });
-  const actionArea = buttons[0].closest('p');
-  actionArea.classList.add('action-area');
-  actionArea.nextElementSibling?.classList.add('supplemental-text', 'body-XL');
 }
 
-export function decorateIcons(el) {
+export function decorateIconArea(el) {
   const icons = el.querySelectorAll('.icon');
   icons.forEach((icon) => {
     icon.parentElement.classList.add('icon-area');
     if (icon.textContent.includes('persona')) icon.parentElement.classList.add('persona-area');
   });
 }
 
+export async function getSVGsfromFile(path, selectors) {
+  if (!path) return null;
+  const resp = await fetch(path);
+  if (!resp.ok) return null;
+
+  const text = await resp.text();
+  const parser = new DOMParser();
+  const doc = parser.parseFromString(text, 'image/svg+xml');
+
+  if (!selectors) {
+    const svg = doc.querySelector('svg');
+    if (svg) return [{ svg }];
+    /* c8 ignore next */
+    return null;
+  }
+
+  return selectors.map((selector) => {
```

Comment:
I think it'd be better to return a dict so that you don't have to loop through all of the svgs when looking them up (on line 76):
```
{
  [selector]: svg,
}
```

Reply:
Also I wonder if it'd be better to just parse all of the icons in the svg file and return that dict?  Then the file is parsed once and you don't have to re-scan the file for every icon block.  You also will not need `getIconList` anymore.

As it is now, the same svgs are re-created for every block.

Reply:
Agreed, this has been updated/refactored.